code_generator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt

```diff
@@ -0,0 +1,85 @@
+
+<#@ template language="C#" #>
+<#@ output extension=".cs" #>
+<#
+    var generatedHeader = CommonGeneratorUtils.GetGeneratedHeader();
+    var componentDetails = GetComponentDetails();
+    var fieldDetailsList = GetFieldDetailsList();
+    var eventDetailsList = GetEventDetailsList();
+    var commandDetailsList = GetCommandDetailsList();
+#>
+<#= generatedHeader #>
+
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using Unity.Entities;
+
+namespace <#= qualifiedNamespace #>
+{
+    public partial class <#= componentDetails.ComponentName #>
+    {
+        public partial class Accessors
+        {
+<# for (var j = 0; j < commandDetailsList.Count; j++) {
+        var commandName = commandDetailsList[j].CommandName;
+        var requestType = commandDetailsList[j].RequestType;
+        var responseType = commandDetailsList[j].ResponseType;
+#>
+            internal class <#= commandName #>CommandSenderCreator : IInjectableCreator
+            {
+                public IInjectable CreateInjectable(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
+                {
+                    return new <#= commandName #>CommandSender(entity, entityManager, logDispatcher);
+                }
+            }
+
+            [InjectableId(InjectableType.CommandSender, <#= unityComponentDefinition.Id #>, <#= j #>)]
```

Comment:
Got used to using `i` in the code and preferring `j` for the codegen meta loop... Might not actually have counted loops in the code here even when it's done though, as far as I can tell. Should I change it to `i`?

Reply:
Oh right - its because there isn't a command index yet. I'd be careful assuming that index in the array = command index.

Once you have the Core merge you'll have the command index

Reply:
Oh I was under the impression that assumption is safe. Making a note on the merge branch for this

Reply:
Actually - I think with the current impl `j != commandIndex` ever.

Because `commandIndex` starts at 1 :trollface:

Reply:
😱